### ISSUE TITLE
iris_depth_from_image_desktop example hangs

### ISSUE THREAD

**Issue by shortcipher3** [google/mediapipe#1033]
I am using the provided Dockerfile and add the following line to build the iris depth example.

```
bazel build -c opt --define MEDIAPIPE_DISABLE_GPU=1 mediapipe/examples/desktop/iris_tracking:iris_depth_from_image_desktop
```

It built fine.  then from the docker image I ran:
```
GLOG_logtostderr=1 bazel-bin/mediapipe/examples/desktop/iris_tracking/iris_depth_from_image_desktop --input_image_path=<input image path> --output_image_path=<output image path>
```
Replacing the image paths with my desired images.

I get some output then it hangs:
```
I20200822 21:16:33.688568    53 iris_depth_from_image_desktop.cc:130] Get calculator graph config contents: # MediaPipe graph that performs iris distance computation on desktop with
# TensorFlow Lite on CPU.
# Used in the example in
# mediapipie/examples/desktop/iris_tracking:iris_depth_from_image_desktop.
...
# Renders annotations and overlays them on top of the input images.
node {
  calculator: "IrisAndDepthRendererCpu"
  input_stream: "IMAGE:input_image"
  input_stream: "FACE_LANDMARKS:face_landmarks"
  input_stream: "EYE_LANDMARKS_LEFT:left_eye_contour_landmarks"
  input_stream: "EYE_LANDMARKS_RIGHT:right_eye_contour_landmarks"
  input_stream: "IRIS_LANDMARKS_LEFT:left_iris_landmarks"
  input_stream: "IRIS_LANDMARKS_RIGHT:right_iris_landmarks"
  input_stream: "NORM_RECT:face_rect"
  input_stream: "LEFT_EYE_RECT:left_eye_rect_from_landmarks"
  input_stream: "RIGHT_EYE_RECT:right_eye_rect_from_landmarks"
  input_stream: "DETECTIONS:face_detections"
  input_side_packet: "IMAGE_FILE_PROPERTIES:image_file_properties"
  output_stream: "IMAGE:output_image"
  output_stream: "LEFT_IRIS_DEPTH_MM:left_iris_depth_mm"
  output_stream: "RIGHT_IRIS_DEPTH_MM:right_iris_depth_mm"
}
I20200822 21:16:33.690336    53 iris_depth_from_image_desktop.cc:136] Initialize the calculator graph.
I20200822 21:16:33.700639    53 iris_depth_from_image_desktop.cc:59] Load the image.
I20200822 21:16:33.707743    53 iris_depth_from_image_desktop.cc:63] Start running the calculator graph.
```

**Comment by shortcipher3** [comment]
It worked today, I can't see anything that I'm doing differently.
```
docker run --rm -ti mediapipe
GLOG_logtostderr=1 bazel-bin/mediapipe/examples/desktop/iris_tracking/iris_depth_from_image_desktop --input_image_path=<input image path> --output_image_path=<output image path>
```

**Comment by shortcipher3** [comment]
It seems that it depends on the image that is provided.  Some images I get:
```
E20200902 03:32:32.318114   256 iris_depth_from_image_desktop.cc:156] Failed to run the graph: Failed to get packet from output stream 'left_iris_depth_mm'.
```

Other images it hangs, and some images it runs just fine.

**Comment by shortcipher3** [comment]
One case where it hangs is on images in which no face is detected.

I've worked on tracking down where it hangs and found that non-max suppression is given no detections, but it sends out a status ok and no output.  The letterbox_removal calculator takes its output to remove the image padding and runs process, this is where it hangs.  If I set the option on non_max_suppression to return an empty vector of Detections then it gets past the letterbox_removal calculator.

Without digging into how all the other pipelines use these nodes, I think the safest fix is to add an option on non_max_suppression that throws an error if the input vector is empty and the option is set. 

I've created a pull request here:
https://github.com/google/mediapipe/pull/1102

Another case where it hangs is when only one eye is visible in the image.  I'm going to start looking closer at that, but I would appreciate feedback on my pull request.
